src/components/permissions/loginHelpers.ts

```diff
@@ -0,0 +1,50 @@
+import databaseConnect from '../databaseConnect'
+import config from '../../config.json'
+import { verify, sign } from 'jsonwebtoken'
+import { promisify } from 'util'
+import { PermissionRow, TemplatePermissions, UserInfo } from './types'
+import { compileJWT } from './rowLevelPolicyHelpers'
+
+const verifyPromise: any = promisify(verify)
+const signPromise: any = promisify(sign)
+
+const getUsername = async (jwtToken: string) => {
+  let username = 'nonRegistered'
+  if (jwtToken) {
+    try {
+      username = (await verifyPromise(jwtToken, config.jwtSecret)).user.username
```

Comment:
```suggestion
      username = (await verifyPromise(jwtToken, config.jwtSecret)).username
```
Think it should be this, username is root level of JWT. (it will be once you re-instate it ;)